cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureStorageAccountService.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import static com.sequenceiq.cloudbreak.cloud.azure.AzureDiskType.LOCALLY_REDUNDANT;
+import static com.sequenceiq.cloudbreak.cloud.azure.AzureStorage.IMAGES_CONTAINER;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.storage.StorageAccount;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.task.AzurePollTaskFactory;
+import com.sequenceiq.cloudbreak.cloud.azure.task.storageaccount.StorageAccountCheckerContext;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
+import com.sequenceiq.cloudbreak.cloud.scheduler.SyncPollingScheduler;
+import com.sequenceiq.cloudbreak.cloud.task.PollTask;
+
+@Component
+public class AzureStorageAccountService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStorageAccountService.class);
+
+    private static final int STORAGE_ACCOUNT_CREATION_CHECKING_INTERVAL = 1000;
+
+    private static final int STORAGE_ACCOUNT_CREATION_CHECKING_MAX_ATTEMPT = 30;
+
+    @Inject
+    private AzurePollTaskFactory azurePollTaskFactory;
+
+    @Inject
+    private SyncPollingScheduler syncPollingScheduler;
+
+    @Inject
+    private AzureStorage armStorage;
+
+    public void createStorageAccount(AuthenticatedContext ac, AzureClient client, String resourceGroup, String storageName, String region, CloudStack stack) {
+        StorageAccount storageAccount = client.getStorageAccountByGroup(resourceGroup, storageName);
+        if (storageAccount == null) {
+            try {
+                LOGGER.info("Creating storage account: {}", storageName);
+                armStorage.createStorage(client, storageName, LOCALLY_REDUNDANT, resourceGroup, region, isEncryptionNeeded(stack), stack.getTags());
+                pollStorageAccountCreation(ac, new StorageAccountCheckerContext(client, resourceGroup, storageName));
+                LOGGER.info("Creating container: {} in storage storage account: {}", IMAGES_CONTAINER, storageName);
+                client.createContainerInStorage(resourceGroup, storageName, IMAGES_CONTAINER);
```

Comment:
pls. use a separate method for this or rename this one to indicate that it is doing this, too